### ISSUE TITLE
can add tensorboardx for visualize the training status?

### ISSUE THREAD

**Issue by mlinxiang** [facebookresearch/maskrcnn-benchmark#76]
## 🚀 Feature
current training information only display in console, this is not very anschauung，can add tensorboardx to support loss or feature map 's visualize

**Comment by fmassa** [comment]
Adding support for TensorBoard would be a good addition, but I'm currently without bandwidth to add it.

Another alternative is to have a script to parse the logs that are printed, and plot them in a jupyter notebook for example (that's what I've been doing).

**Comment by steve-goley** [comment]
I'm planning to add a visdom plotter. Also the only feature I miss from TF OD API is the concurrent eval loop. That's also on my list.

**Comment by fmassa** [comment]
@steve-goley what do you mean by concurrent eval loop?
Currently, it is possible to evaluate using multiple GPUs, is that what you meant?

**Comment by fmassa** [comment]
@steve-goley oh, I see, yes, that would definitely be a plus to have!

**Comment by mlinxiang** [comment]
@steve-goley can provide any docs describe TF OD API's visualize support mechanism，so may i can add some doc to do this

**Comment by steve-goley** [comment]
@mlinxiang TensorFlow uses TensorBoard. I haven't dug into the checkpoint structure, yet. I think that a similar capability could be constructed with visdom. There are some advantages and disadvantages to visdom over Tensorboard / tensorboardX. I've found visdom quite easy to use and the API, much like many things FAIR, is quite readable and concise. @fmassa do you have a strong opinion on tensorboardX vs. visdom.

**Comment by soumith** [comment]
given the rich nature and potential of mask-rcnn visualizations, where a lot of custom visualizations might make sense, Visdom seems better longer-term. Visdom has trivial ways to construct rich custom visualizations.

**Comment by mlinxiang** [comment]
has visdom integrate into pytorch?

**Comment by fmassa** [comment]
I think Visdom works with PyTorch

**Comment by wangg12** [comment]
I think using tensorboardX to visualize the accuracies and losses in real time is a good idea.
